tracking-migration/src/main/java/io/customer/tracking/migration/queue/QueueRunRequest.kt

```diff
@@ -14,52 +14,61 @@ internal class QueueRunRequestImpl internal constructor(
 ) : QueueRunRequest {
 
     override suspend fun run() {
-        logger.debug("queue starting to run tasks...")
-        val inventory = queueStorage.getInventory()
-        val tasksToRun = inventory.toMutableList()
+        runCatching {
+            logger.debug("queue starting to run tasks...")
+            val tasksToRun = queueStorage.getInventory().toMutableList()
+            while (tasksToRun.isNotEmpty()) {
+                // get the next task to run using the query criteria
+                val currentTaskMetadata = queryRunner.getNextTask(tasksToRun)
+                if (currentTaskMetadata == null) {
+                    logger.debug("all queue tasks have been migrated or failed to run. Exiting queue run.")
+                    break
```

Comment:
codecov points out we are not testing the `break` case? it would be good to add it

Reply:
Added test case to cover this line too